backup-stores/s3/src/test/java/io/camunda/zeebe/backup/s3/BackupUploadIT.java

```diff
@@ -0,0 +1,171 @@
+
+  @Container
+  private static final LocalStackContainer S3 =
+      new LocalStackContainer(DockerImageName.parse("localstack/localstack"))
+          .withServices(Service.S3)
+          .withEnv("LS_LOG", "trace");
+
+  @RegisterExtension
+  final ContainerLogsDumper logsWatcher = new ContainerLogsDumper(() -> Map.of("localstack", S3));
+
+  private S3AsyncClient client;
+  private S3BackupStore store;
+  private S3BackupConfig config;
+
+  public S3AsyncClient getClient() {
+    return client;
+  }
+
+  public S3BackupConfig getConfig() {
+    return config;
+  }
+
+  public S3BackupStore getStore() {
+    return store;
+  }
+
+  @BeforeAll
+  static void setupBucket() {
+    final var config =
+        new Builder()
+            .withBucketName(BUCKET_NAME)
+            .withEndpoint(S3.getEndpointOverride(Service.S3).toString())
+            .withRegion(S3.getRegion())
+            .withCredentials(S3.getAccessKey(), S3.getSecretKey())
+            .build();
+    try (final var client = S3BackupStore.buildClient(config)) {
+      client.createBucket(CreateBucketRequest.builder().bucket(config.bucketName()).build()).join();
+    }
+  }
+
+  @AfterEach
+  void tearDown() {
+    store.closeAsync();
+  }
+
+  public void setConfigParallelConnectionsAndTimeout(
+      final int parallelConnections, final Duration timeout) {
+    config =
+        new Builder()
+            .withBucketName(BUCKET_NAME)
+            .withBasePath(RandomStringUtils.randomAlphabetic(10).toLowerCase())
+            .withEndpoint(S3.getEndpointOverride(Service.S3).toString())
+            .withRegion(S3.getRegion())
+            .withCredentials(S3.getAccessKey(), S3.getSecretKey())
+            .withApiCallTimeout(null)
+            .forcePathStyleAccess(false)
+            .withCompressionAlgorithm(null)
+            .withConnectionAcquisitionTimeout(timeout)
+            .withParallelUploadsLimit(parallelConnections)
+            .build();
+    client = S3BackupStore.buildClient(config);
+    store = new S3BackupStore(config, client);
+  }
+
+  @Test
+  void shouldSaveBackupWithManyFiles() throws IOException {
+    // given
+    // Default values for the configuration
+    setConfigParallelConnectionsAndTimeout(50, Duration.ofSeconds(10));
+    final CompletableFuture<Void> saveFuture = getStore().save(largeNumberOfSegmentsBackup(4_000));
+
+    // then
+    Assertions.assertThat(saveFuture).succeedsWithin(Duration.ofSeconds(60));
+  }
+
+  @Test
+  void shouldNotTimeoutForAcquisitionOfConnection() throws IOException {
+    // given
+    // Even with just one connection, and low timeout limit, the second upload should not start
+    // until a connection is available, and therefore should not throw AcquisitionConnectionTimeout
+    setConfigParallelConnectionsAndTimeout(1, Duration.ofMillis(50));
+    final CompletableFuture<Void> saveFuture = getStore().save(configurableLargeBackup(500_000));
```

Comment:
:thought_balloon: The files are just 500KB and I'm not sure this test is reliable enough. I'm fine merging as is but I think it could also be a nice opportunity to make use of [toxiproxy](https://java.testcontainers.org/modules/toxiproxy/) to instead introduce a lot of latency between or store and the s3 container. Let me know if you are interested in that, we could pair on it.

Reply:
I increased the file size to 50MB, which also works with larger files but is quite long. This module seems interesting, do we already use it somewhere in the code?